rust-runtime/smithy-types/src/retry.rs

```diff
@@ -149,6 +149,46 @@ impl Default for RetryConfig {
     }
 }
 
+#[derive(Debug)]
+pub enum RetryConfigErr {
+    InvalidRetryMode {
+        source: RetryModeErr,
+        set_by: String,
+    },
+    MaxAttemptsMustNotBeZero {
+        set_by: String,
+    },
+    AdaptiveModeIsNotSupported {
+        set_by: String,
+    },
+}
+
+impl Display for RetryConfigErr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        use RetryConfigErr::*;
+        match self {
+            InvalidRetryMode { set_by, source } => {
+                write!(f, "invalid configuration set by {}: {}", set_by, source)
+            }
+            MaxAttemptsMustNotBeZero { set_by } => {
+                write!(f, "invalid configuration set by {}: It is invalid to set max attempts to 0. Unset it or set it to an integer greater than or equal to one.", set_by)
+            }
+            AdaptiveModeIsNotSupported { set_by } => {
+                write!(f, "invalid configuration set by {}: Setting retry mode to 'adaptive' is not yet supported. Unset it or set it to a supported mode.", set_by)
```

Comment:
can you list supported modes here?

Reply:
legacy is also unsupported, fwiw

Reply:
I was covering `legacy` with the `InvalidRetryMode(RetryModeError)` error where it would come up as invalid. I added the special case for adaptive because we do intend to support it, just not yet. Would you like me to do either of:
1. create a special case for `legacy`. it's a not-uncommon AWS settings for other SDKs and people should get a more helpful message
2. remove the special case for `adaptive` and just say that anything other than `standard` is invalid
3. Another option I hadn't considered